### PR TITLE
chore: Update Checkbox and Radio to use griffel reset styles

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Checkbox and Radio's styles include all of the styles for the checked, hover, etc. states in the base styles, and use CSS selectors to conditionally apply the styles.
* This results in a large number of classes being added to the default checkbox, because Griffel styles use a separate class for each combination of selector & property. As shown below, there are a total of **154** css classes across all of Checkbox's slots.
* This affects performance because chromium's css performance degrades with the number of classes.

Checkbox class counts

## New Behavior

Use Griffel's `makeResetStyles` to combine all of the base styles for each of Checkbox and Radio's slots into a single CSS class. Also, merge the styles for the default state into the base styles (`size: medium`, `labelPosition: after`). 
* For Checkbox, this results in an **85%** reduction in the number of CSS classes across all slots: down to **23** from 154 before.
* The stress test for Checkbox shows a roughly **12%** improvement in performance for applying styling.

Checkbox class counts

Checkbox perf test results

## Related Issue(s)

* Partial fix for: #25483
  * This fixes the **CSS class bloat** part but does not fix the **Override specificity** concern in that issue.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `59.288 kB`<br />`16.435 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `186.264 kB`<br />`52.303 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.75 kB`<br />`11.101 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/baf93e56255ce8136a965a7304d7d5bcc0042384'>baf93e56255ce8136a965a7304d7d5bcc0042384</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Avatar-mount_base.html">1324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Avatar-mount.html">1337</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Button-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Button-mount.html">888</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProvider-mount_base.html">1562</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProvider-mount.html">1649</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-mount_base.html">642</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-mount.html">631</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">597</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">599</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">555</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">624</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/MakeStyles-mount_base.html">1933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/MakeStyles-mount.html">1880</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Persona-mount_base.html">3040</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/Persona-mount.html">2964</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/SpinButton-mount_base.html">2527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25984/perf-test-react-components/SpinButton-mount.html">2311</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts [hunk 1]
```diff
@@ -15,174 +15,169 @@ export const checkboxClassNames: SlotClassNames<CheckboxSlots> = {
...
+  alignSelf: 'flex-start',
+  boxSizing: 'border-box',
+  flexShrink: 0,
+
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  overflow: 'hidden',
+
+  borderWidth: tokens.strokeWidthThin,
```
Comment (spmonahan) [inline packages/react-components/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts:168]: nit: I think this line and the next can be collapsed into

```ts
border: `${tokens.strokeWidthThin} solid`
```

**Review by spmonahan — APPROVED** [review]

**Pull request merged**